moonshot/src/bookmark/bookmark.py

```diff
@@ -0,0 +1,235 @@
+                    f"[Bookmark] No record found for bookmark name {bookmark_name}"
+                )
+        else:
+            raise RuntimeError(f"[Bookmark] Invalid bookmark name: {bookmark_name}")
+
+    def delete_bookmark(self, bookmark_name: str) -> dict:
+        """
+        Delete a bookmark by its unique name.
+
+        Args:
+            bookmark_name (str): The unique name for the bookmark to be deleted.
+        """
+        if bookmark_name is not None:
+            try:
+                sql_delete_bookmark_record = f"""
+                    DELETE FROM bookmark WHERE name = '{bookmark_name}';
+                """
+                Storage.delete_database_record_in_table(
+                    self.db_instance, sql_delete_bookmark_record
+                )
+                return {"success": True, "message": "Bookmark record deleted."}
+            except Exception as e:
+                error_message = f"Failed to delete bookmark record: {e}"
+                return {"success": False, "message": error_message}
+        else:
+            error_message = f"[Bookmark] Invalid bookmark name: {bookmark_name}"
+            return {"success": False, "message": error_message}
+
+    def delete_all_bookmark(self) -> dict:
+        """
+        Delete all bookmarks from the database and return the operation result.
+
+        Returns:
+            dict: A dictionary with 'success' status and 'message' containing an error message if failed.
+        """
+        try:
+            Storage.delete_database_record_in_table(
+                self.db_instance, Bookmark.sql_delete_bookmark_records
+            )
+            return {"success": True, "message": "All bookmark records deleted."}
+        except Exception as e:
+            error_message = f"Failed to delete all bookmark records: {e}"
+            return {"success": False, "message": error_message}
+
+    def export_bookmarks(self, export_file_name: str = "bookmarks") -> str:
+        """
+        Export all bookmarks to a JSON file.
+
+        This method retrieves all bookmarks from the database, converts them to a JSON format,
+        and writes them to a file in the 'moonshot-data/bookmark' directory with the provided file name.
+
+        Args:
+            export_file_name (str): The base name of the file to export the bookmarks to.
+                                    The '.json' extension will be appended to this base name.
+
+        Returns:
+            str: The path to the exported JSON file containing the bookmarks.
+        """
+        list_of_bookmarks_tuples = Storage.read_database_records(
+            self.db_instance,
+            Bookmark.sql_select_bookmarks_record,
+        )
+
+        bookmarks_json = [
+            BookmarkArguments.from_tuple_to_dict(bookmark_tuple)
+            for bookmark_tuple in list_of_bookmarks_tuples
+        ]
+
+        # Write json file to moonshot-data
+        file_path = f"../moonshot-data/bookmark/{export_file_name}.json"
```

Comment:
would prefer to have the filepath not hardcoded here, but in class itself or somewhere above, either by self or class level.

Reply:
can we read the location to write to from the Storage get bookmark env var?

Reply:
this one we will update once kelvin merge the changes for storage .py